Source/Internal/UICollectionViewLayout+InteractiveReordering.m

```diff
@@ -60,7 +61,9 @@ + (void)load
 }
 
 - (void)ig_hijackLayoutInteractiveReorderingMethodForAdapter:(IGListAdapter *)adapter {
-    objc_setAssociatedObject(self, kIGListAdapterKey, adapter, OBJC_ASSOCIATION_ASSIGN);
```

Comment:
Maybe the problem here is the `OBJC_ASSOCIATION_ASSIGN`, can you try using the `OBJC_ASSOCIATION_RETAIN_NONATOMIC` ?  

IGListAdapter holds a weak reference to UICollectionView, so maybe we are fine here. Just make sure we dont get into a retain cycle issue, then we can avoid using a IGWeakObject here.